File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/sqlserver/NullableTypes.scala
package dev.typr.foundations.docs.sqlserver
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*

@SuppressWarnings(Array("unused"))
object NullableTypes:
  //start
  val notNull: SqlServerType[Int] = SqlServerTypes.int_
  val nullable: SqlServerType[Option[Int]] = SqlServerTypes.int_.nullable
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/sqlserver/NullableTypes.java
package dev.typr.foundations.docs.sqlserver;

import dev.typr.foundations.SqlServerType;
import dev.typr.foundations.SqlServerTypes;
import java.util.Optional;

@SuppressWarnings("unused")
public class NullableTypes {
  // start
  SqlServerType<Integer> notNull = SqlServerTypes.int_;
  SqlServerType<Optional<Integer>> nullable = SqlServerTypes.int_.opt();
  // stop
}
